### ISSUE TITLE
Missing libsteam_api

### ISSUE THREAD

**Issue by Kamiikaze** [FezVrasta/ark-server-tools#1063]
Since the latest Update (Build 4715767) my Server doesnt start anymore. Telling me 
`ShooterGameServer: error while loading shared libraries: libsteam_api.so: cannot open shared object file: No such file or directory`

**Comment by Nytridr** [comment]
I am getting the same thing..
2020-02-26 00:00:50: [Ragnarok] /home/steam/ARK/ShooterGame/Binaries/Linux/ShooterGameServer: error while loading shared libraries: libsteam_api.so: cannot open shared object file: No such file or directory

happens with all servers though.. so right now none of my servers are up.. not a good night for that one.

**Comment by fusselshu** [comment]
its a wildcard thing

**Comment by Kamiikaze** [comment]
seems WC Devs messed up something, I think we can only wait.
Because on my second root servers I got Windows running and a running Genesis server, this issue is only for linux version

**Comment by daevien** [comment]
Yeah, it's happened before and isn't an issue with arkmanager I don't believe, just Wildcard breaking linux as usual. Once windows mostly going they might look again at linux branch.

**Comment by bakely** [comment]
that's a rip... I knew i should have just keep playing on the old build... but i terminated it to update it and now i get to deal with this. i guess i won't be playing ark anymore tonight. I feel especially sorry for the other people who play on the server

**Comment by fusselshu** [comment]
There is a workaround for this: https://survivetheark.com/index.php?/forums/topic/507590-error-linux-server-missing-file-not-starting/&do=findComment&comment=2792844

**Comment by daevien** [comment]
Fix that I've confirmed: If you have a backup of the Engine/Binaries/Linux folder, you can copy it back into the location & it works.

**Comment by klightspeed** [comment]
https://steamdb.info/patchnotes/4715767/

They deleted a number of the Linux server directories:
* `Engine/Binaries/Linux`
* `Engine/Config/Linux`
* `ShooterGame/Config/Linux`

Fortunately the latter 2 seem to only apply to the Linux client

They also deleted version.txt

You might be able to try copying the `Engine/Binaries/Linux` directory from a server that hasn't been updated yet.

If you don't have that, here is a tar.gz of the Engine/Binaries/Linux directory on a server I last updated in January:
[ARKServer-Engine-Binaries-Linux.tar.gz](https://github.com/FezVrasta/ark-server-tools/files/4253887/ARKServer-Engine-Binaries-Linux.tar.gz)

**Comment by Nytridr** [comment]
Just incase this will help someone.. this is just what I did.. I just created a symlink and it worked.

ln -s /home/steam/linuxgsm/serverfiles/Engine/Binaries/Linux/ /home/steam/ARK/Engine/Binaries/Linux

**Comment by Nytridr** [comment]
okay the above worked for my other servers, (but patch just came out so... )   would [S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed. be part of this or something else?

**Comment by klightspeed** [comment]
`[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.` is benign - the steam API has always given this error when the Steam client isn't running.

**Comment by Kamiikaze** [comment]
> https://steamdb.info/patchnotes/4715767/
> 
> They deleted a number of the Linux server directories:
> 
> * `Engine/Binaries/Linux`
> * `Engine/Config/Linux`
> * `ShooterGame/Config/Linux`
> 
> Fortunately the latter 2 seem to only apply to the Linux client
> 
> They also deleted version.txt
> 
> You might be able to try copying the `Engine/Binaries/Linux` directory from a server that hasn't been updated yet. Edit: @daevien beat me to it.
> 
> If you don't have that, here is a tar.gz of the Engine/Binaries/Linux directory on a server I last updated in January:
> [ARKServer-Engine-Binaries-Linux.tar.gz](https://github.com/FezVrasta/ark-server-tools/files/4253887/ARKServer-Engine-Binaries-Linux.tar.gz)

Thank you, works for me! :)

**Comment by bakely** [comment]
> https://steamdb.info/patchnotes/4715767/
> 
> They deleted a number of the Linux server directories:
> 
> * `Engine/Binaries/Linux`
> * `Engine/Config/Linux`
> * `ShooterGame/Config/Linux`
> 
> Fortunately the latter 2 seem to only apply to the Linux client
> 
> They also deleted version.txt
> 
> You might be able to try copying the `Engine/Binaries/Linux` directory from a server that hasn't been updated yet. Edit: @daevien beat me to it.
> 
> If you don't have that, here is a tar.gz of the Engine/Binaries/Linux directory on a server I last updated in January:
> [ARKServer-Engine-Binaries-Linux.tar.gz](https://github.com/FezVrasta/ark-server-tools/files/4253887/ARKServer-Engine-Binaries-Linux.tar.gz)

after doing this... my servers settings were changed like harvest rate

**Comment by klightspeed** [comment]
Different people with different development snapshots must be pushing updates.

In buildid [4715767](https://steamdb.info/patchnotes/4715767/), Genesis was released, with Linux libraries, Primitive Plus, and version.txt being removed.
In buildid [4719750](https://steamdb.info/patchnotes/4719750/), they re-added the missing libraries.
In buildid [4719814](https://steamdb.info/patchnotes/4719814/), they re-added Primitive Plus.
In buildid [4724987](https://steamdb.info/patchnotes/4724987/), they re-added version.txt and some of the missing Linux files, and the capitalisation on PumpkinHead/Icon was fixed
In buildid [4725556](https://steamdb.info/patchnotes/4725556/), then re-removed version.txt and some of the missing Linux files, and the capitalisation on PumpkinHead/ICon was un-fixed.

> after doing this... my servers settings were changed like harvest rate

@bakely is this the first time it has successfully run since updating to Genesis?

If you compare the current `GameUserSettings.ini` and `Game.ini` to those in a recent backup from before Genesis landed, are there any differences?

**Comment by Alphakilo** [comment]
Symlinked directory will break the loading of shared objects by `ShooterGameServer`. To debug this, please run the failing commandline with strace, e.g. `strace -e file /home/steam/ARK/ShooterGame/Binaries/Linux/ShooterGameServer ...`
